### ISSUE TITLE
Using @container query logs error

### ISSUE THREAD

**Issue by zbigg** [jsdom/jsdom#3597]
### Basic info:

- **Node.js version:** v20.6.1
- **jsdom version:** 22.1.0

### Minimal reproduction case

```js
const { JSDOM } = require('jsdom');

new JSDOM(`
  <style>
    @container (min-width: 700px) {
      .card h2 {
        font-size: 2em;
      }
    }
  </style>
`);
```

The container query is taken from [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_container_queries) in this feature, so it should be correct (?)

Sorry if there is duplicate already, but brief search was unsuccessfull.

### How does similar code behave in browsers?

Works as expected - container query has effect on style applied to elements.

**Comment by zbigg** [comment]
Looks like this is implemented by `cssom` and there is open issue there: https://github.com/NV/CSSOM/issues/109 and not being tackled  `rrweb-ccsom` used currently by `jsdom`.

**Comment by domenic** [comment]
I don't think this is implemented by cssom? We switched to rrweb-cssom because cssom was not maintained.

**Comment by zbigg** [comment]
And sorry for wrong title (fixed now), it just reports error on virtual console and classic workaround of providing custom virtual console (possibly silent):

```
new jsdom.JSDOM('',
  { virtualConsole: new jsdom.VirtualConsole() }
)

**Comment by hoepfnerdennis** [comment]
There is the same error log when using `@layer`...
```css
const { JSDOM } = require('jsdom');

new JSDOM(`
  <style>
    @layer {
      .card h2 {
        font-size: 2em;
      }
    }
  </style>
`);
```

**Comment by dustindowns** [comment]
Getting the same `@layer` error too when attempting to test with PrimeVue components.

<img width="1381" alt="Screenshot 2023-10-13 at 2 16 04 PM" src="https://github.com/jsdom/jsdom/assets/2736672/04dbc749-bb98-42af-9a64-2ea44e80afeb">

**Comment by viceice** [comment]
any known workaround for jest jsdom? I'm facing the issue with primeng and `@layer`.

**Comment by cluxig** [comment]
Same for me. Unable to test primeng properly.

**Comment by SergioCanalejo** [comment]
Any update or workaround about that? I've been stuck in a old primevue version because of that

**Comment by jantimon** [comment]
in the meantime @domenic and @seanparmelee switched from cssom 0.5.0 to rrweb-cssom 0.6.0 in #3497  
rrweb-cssom is maintained by @YunFeng0817 [npm/fitz5264](https://www.npmjs.com/~fitz5264)

so I moved the `@container` fix by @romain-blatrix there: https://github.com/rrweb-io/CSSOM/pull/2

lets hope for the best

**Comment by alopix** [comment]
It has already been merged. Now to hope for a new version being released someday.

**Comment by YunFeng0817** [comment]
Hi jsdom team, v0.7.0 has been published!

**Comment by jantimon** [comment]
thanks @YunFeng0817

I created a pr to bump the version:

https://github.com/jsdom/jsdom/pull/3707/files

**Comment by jantimon** [comment]
Adding support `@container` fixes a test of the JSDOM test suite

This sounds good however the test does also other things which works only for Node 20 but fails for Node 18

So unfortunately it looks like that the JSDOM test setup will block `@container` support for JSDOM

More details here: #3707

**Comment by jantimon** [comment]
finally the pipeline is green 🎉

**Comment by domenic** [comment]
I'll close this since I guess people are happy we no longer throw errors about container queries. But we don't actually implement them correctly, so, I'll open a new issue tracking that, and add a test confirming that our implementation just ignores them.

In the future, people who want to ignore CSS and just get no errors should pay attention to #2005.
